hazelcast/src/main/java/com/hazelcast/security/permission/InstancePermission.java

```diff
@@ -66,11 +66,7 @@ public boolean implies(Permission permission) {
             return false;
         }
 
-        if (!CONFIG_PATTERN_MATCHER.matches(that.getName(), getName())) {
-            return false;
-        }
-
-        return true;
+        return CONFIG_PATTERN_MATCHER.matches(getName(), that.getName());
```

Comment:
the arguments were flipped. 
we want to see if I can access a structure "map.foo" if my permission says I can access "map.*". not the other way around.